mise completion clink fails when user path contains spaces

Running the command:
`
mise completion clink
`
produces the following output:
`
mise completion clink
Generating completions ...
'C:\Users\Binyamin' is not recognized as an internal or external command,
operable program or batch file.
[ERROR] failed to generate completions (path:C:\ProgramData\clink\mise-clink\mise.usage.lua); exit code: 255
`
Steps to Reproduce:

1.     Install mise on Windows in a user profile path containing spaces.
       (Example: C:\Users\Binyamin Blatt)

2.    Run:
`
mise completion clink
`
Expected Result:
Completions should be generated successfully regardless of spaces in the user profile path.

Actual Result:
The command fails with:
`
'C:\Users\Binyamin' is not recognized as an internal or external command
`
indicating the executable path is not properly quoted.

Environment:

    OS: Windows 10/11

    mise version: 2025.7.27 windows-x64 (2025-07-24)

    Clink version: 1.7.21.9f5af8

    Shell: cmd.exe (via Clink)

Possible Cause:
The `mise completion clink` command likely calls an internal process without quoting the full path to the mise executable, causing the Windows command interpreter to split it at the space in the username.

Suggested Fix:
Wrap all executable paths in double quotes when invoking them from the completion generator.

--- a/modules/cuc.lua
+++ b/modules/cuc.lua
@@ -35,7 +35,7 @@ function Cuc:download_cuc()
 end
 
 function Cuc:generate_completions(mise_path)
-    local cmd_line = string.format([[%s usage | %s generate --complete]], mise_path, self.path)
+    local cmd_line = string.format([[""%s" usage | "%s" generate --complete"]], mise_path, self.path)
     local p = io.popen(cmd_line)
     assert(p, "[ERROR] failed to generate completions: " .. (cmd_line or "(nil)"))
     local output = p:read("*a")

